### PR TITLE
[WIP][ISSUE #482]Support eventmesh server plugin

### PR DESCRIPTION
**Pull request by ruanwenjun**

### CODE REVIEW

**Comment by Jackzeng1224** [comment]
@ruanwenjun   hello,There was a problem when verifying the version. See the picture. I have added dependencies for related configurations.
image
image

**Comment by ruanwenjun** [comment]
@Jackzeng1224 I can run the test in AsyncPublishInstance, could you please clean the local cache class, and test again.
image

**Comment by Jackzeng1224** [comment]
> @Jackzeng1224 I can run the test in AsyncPublishInstance success, could you please clean the local cache class, and test again.
> image

@ruanwenjun     Every time I work on a new project. Now the newly downloaded project has a new problem.
image

**Comment by ruanwenjun** [comment]
@Jackzeng1224 This may be caused by the TCP Server plugin have not been installed.
You can add this dependency in build.gradle, or remove the tcp in eventmesh.yml.
```
implementation project(":eventmesh-protocol-plugin:eventmesh-protocol-tcp")
```

I will optimize this tonignt

**Comment by Jackzeng1224** [comment]
> @Jackzeng1224 This may be caused by the TCP Server plugin have not been installed.
> You can add this dependency in build.gradle, or remove the tcp in eventmesh.yml.
> 
> ```
> implementation project(":eventmesh-protocol-plugin:eventmesh-protocol-tcp")
> ```
> 
> I will optimize this tonignt

ok.I add it and try.    
I have a question.Why can't I configure one separately? If there are 3 or 4 ways, but I only want to use one.
image

**Comment by ruanwenjun** [comment]
@Jackzeng1224 Of course you can. You can config use which plugin in eventmesh.yml.
You can remove the tcp and then you can just use http plugin
image

**Comment by Jackzeng1224** [comment]
@ruanwenjun    hello， Has this protocol adapter plugin been submitted?

**Comment by ruanwenjun** [comment]
@Jackzeng1224 Yes, I have rebased the latest code.

**Comment by Jackzeng1224** [comment]
> @Jackzeng1224 Yes, I have rebased the latest code.

OK.

**Comment by Jackzeng1224** [comment]
@ruanwenjun 
hello.  I tried the tcp plugin mode, cannot find the protocol plugin: tcp.
image
image

**Comment by ruanwenjun** [comment]
@Jackzeng1224 When you add the dependency in `build.gradle`, you need to refresh the gradle.
And you can also use ./gradlew clean jar dist to install the plugins.

**Comment by Jackzeng1224** [comment]
> @Jackzeng1224 When you add the dependency in `build.gradle`, you need to refresh the gradle.
> And you can also use ./gradlew clean jar dist to install the plugins.

ok.I will try again.

**Comment by ruanwenjun** [comment]
@Jackzeng1224 Do you meet any other problem?

**Comment by Jackzeng1224** [comment]
> @Jackzeng1224 Do you meet any other problem?

Not yet, the functions can be used normally.

**Comment by ruanwenjun** [comment]
@Jackzeng1224 Thanks for your kind assistance.

File: eventmesh-protocol-plugin/eventmesh-protocol-api/build.gradle [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+dependencies {
+    api project(":eventmesh-connector-plugin:eventmesh-connector-api")
```
Comment (xwm1992) [inline eventmesh-protocol-plugin/eventmesh-protocol-api/build.gradle:19]: these two dependencies `connector-api` and `registry-api`  may be change to `":eventmesh-spi"` and  `":eventmesh-common"`

**Review by xwm1992 — CHANGES_REQUESTED** [review]

File: eventmesh-protocol-plugin/eventmesh-protocol-http/build.gradle [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+dependencies {
+    api project(":eventmesh-protocol-plugin:eventmesh-protocol-api")
+    api project(":eventmesh-security-plugin:eventmesh-security-api")
```
Comment (xwm1992) [inline eventmesh-protocol-plugin/eventmesh-protocol-http/build.gradle:20]: can we split the security plugin from the http protocol , integrate the plugin on runtime

**Comment by xwm1992** [comment]
this pr current implementation just separate the server to `http server plugin` and `tcp server plugin` maybe call server plugin more appropriate, not protocol plugin. I think we need to design a `protocol layer api`(tcp and http) which is depended by `server plugin`, to make the protocol layer more extensible.

**Comment by ruanwenjun** [comment]
@xwm1992 @Jackzeng1224 Yes, the module has been renamed to eventmesh-server-plugin, and I will try to design the protocol plugin, maybe this pr can be merged, since the other pr will be blocked.

**Comment by qqeasonchen** [comment]
I think protocol adapter is more needed.

**Comment by ruanwenjun** [comment]
Closed by too many conflicts, if this is needed, we can submit a new pr.

**Pull request closed without merging**